File: VB/CustomFilterPopupExample/ViewModel/TaskViewModel.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace CustomFilterPopupExample
	Friend Class TaskViewModel
		Private _List As ObservableCollection(Of Task)

		Public ReadOnly Property List() As ObservableCollection(Of Task)
			Get
				If _List Is Nothing Then
					_List = New ObservableCollection(Of Task)()
					For i As Integer = 0 To 99
						_List.Add(New Task() With {.Name = "Task" & i, .Date = New DateTime(2014, 10, New Random().Next(1, 31)), .SubscribeNumber = i * i})
					Next i
				End If
				Return _List
			End Get
		End Property

		Public Class Task
			Private privateName As String
			Public Property Name() As String
				Get
					Return privateName
				End Get
				Set(ByVal value As String)
					privateName = value
				End Set
			End Property
			Private privateDate As DateTime
            Public Property [Date]() As DateTime
                Get
                    Return privateDate
                End Get
                Set(ByVal value As DateTime)
                    privateDate = value
                End Set
            End Property
			Private privateSubscribeNumber As Integer
			Public Property SubscribeNumber() As Integer
				Get
					Return privateSubscribeNumber
				End Get
				Set(ByVal value As Integer)
					privateSubscribeNumber = value
				End Set
			End Property
		End Class
	End Class
End Namespace

File: CS/CustomFilterPopupExample/ViewModel/TaskViewModel.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CustomFilterPopupExample
{
    class TaskViewModel
    {
        private ObservableCollection<Task> _List;

        public ObservableCollection<Task> List
        {
            get
            {
                if (_List == null)
                {
                    _List = new ObservableCollection<Task>();
                    for (int i = 0; i < 100; i++)
                    {
                        _List.Add(new Task() { Name = "Task" + i, Date = new DateTime(2014, 10, new Random().Next(1, 31)), SubscribeNumber = i * i });
                    }
                }
                return _List;
            }
        }

        public class Task
        {
            public string Name { get; set; }
            public DateTime Date { get; set; }
            public int SubscribeNumber { get; set; }
        }
    }
}
